### PR TITLE
jsonnet: make memberlist the default KV store

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR changes default value of `memberlist_ring_enabled` to `true`. It means that all hash rings use Memberlist as default KV store instead of Consul, which was the previous default, and Consul is not deployed into the cluster anymore.

Previous "test-gossip*" tests were modified to test use of consul instead.

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/mimir-tests/test-autoscaling-generated.yaml [hunk 1]
```diff
@@ -776,15 +404,16 @@ spec:
...
-        - -distributor.ring.store=consul
-        - -ingester.ring.consul.hostname=consul.default.svc.cluster.local:8500
+        - -distributor.ring.store=memberlist
         - -ingester.ring.heartbeat-timeout=10m
         - -ingester.ring.prefix=
         - -ingester.ring.replication-factor=3
-        - -ingester.ring.store=consul
+        - -ingester.ring.store=memberlist
         - -mem-ballast-size-bytes=1073741824
+        - -memberlist.abort-if-join-fails=false
```
Comment (pracucci) [inline operations/mimir-tests/test-autoscaling-generated.yaml:414]: Not in this PR: should we change the default to false, given we recommend to disable it?

**Review by pracucci — APPROVED** [review]

File: operations/mimir-tests/test-autoscaling-generated.yaml (hunk 1 above)
Comment (pstibrany) [reply operations/mimir-tests/test-autoscaling-generated.yaml:414]: 👍 Let's do that.

Comment (pstibrany) [reply operations/mimir-tests/test-autoscaling-generated.yaml:414]: https://github.com/grafana/dskit/pull/176

**Comment by pstibrany** [comment]
Merging this broke tests as I haven't rebased it on top of `main` and there was a conflicting change.

https://github.com/grafana/mimir/pull/2160 fixes it.

**Pull request merged**